velocity/src/main/java/com/vexsoftware/votifier/velocity/VelocityBackendServer.java

```diff
@@ -0,0 +1,42 @@
+package com.vexsoftware.votifier.velocity;
+
+import com.velocitypowered.api.proxy.ProxyServer;
+import com.velocitypowered.api.proxy.ServerConnection;
+import com.velocitypowered.api.proxy.server.ServerInfo;
+import com.vexsoftware.votifier.platform.BackendServer;
+
+import java.util.Optional;
+
+class VelocityBackendServer implements BackendServer {
+    private final ProxyServer server;
+    private final ServerInfo info;
+
+    VelocityBackendServer(ProxyServer server, ServerInfo info) {
+        this.server = server;
+        this.info = info;
+    }
+
+    @Override
+    public String getName() {
+        return info.getName();
+    }
+
+    @Override
+    public boolean sendPluginMessage(String channel, byte[] data) {
+        Optional<ServerConnection> connection = server.getAllPlayers().stream()
+                .map(p -> p.getCurrentServer().filter(s -> s.getServerInfo().equals(info)))
+                .filter(Optional::isPresent)
+                .findAny()
+                .flatMap(o -> o);
```

Comment:
am i missing something?

Reply:
We get an `Optional<Optional<ServerConnection>>` here, this is simply the identity function to unpack it.